GH-2262: As a smith I want to change when/how tests of n4js-libs are executed and restructure test bundles

mor-n4 opened the pull request:
See #2262.

mor-n4 commented:
Note to self: before merging, compare number of tests on this branch with that on `master`!

mmews-n4 reviewed (approved):
really nice clean-ups!

mor-n4 commented:
Thanks @mmews-n4!

Builds are green and the exact same number of tests are executed on this branch as on master (57150), which is expected because I have deleted 1 test and added 1 test.

Merging ...
